ui/packages/app/src/project_setting/metadata/MetadataList.js

```diff
@@ -0,0 +1,190 @@
+  const [streamError, setStreamError] = useState("");
+  const [isValidStream, setIsValidStream] = useState(false);
+  const onStreamChange = selectedStream => {
+    let isValid = isDNS1123Label(selectedStream);
+    if (!isValid) {
+      setStreamError(
+        "Stream name is invalid. It should contain only lowercase alphanumeric and dash (-)"
+      );
+    }
+    setIsValidStream(isValid);
+    setStream(selectedStream);
+    setTeam(undefined);
+  };
+
+  const teamOptions = useMemo(() => {
+    return (config.STREAMS[selectedStream] || [])
+      .sort((a, b) => a.localeCompare(b))
+      .map(team => ({ label: team.trim() }));
+  }, [selectedStream]);
+
+  const [selectedTeam, setTeam] = useState(project.team);
+  const [teamError, setTeamError] = useState("");
+  const [isValidTeam, setIsValidTeam] = useState(false);
+  const onTeamChange = selectedTeam => {
+    if (selectedTeam !== project.team) {
+      let isValid = isDNS1123Label(selectedTeam);
+      if (!isValid) {
+        setTeamError(
+          "Team name is invalid. It should contain only lowercase alphanumeric and dash (-)"
+        );
+      }
+      setIsValidTeam(isValid);
+      setTeam(selectedTeam);
+    }
+    if (selectedTeam === project.team && selectedStream === project.stream) {
+      setIsValidTeam(false);
+      setIsValidStream(true);
+      setTeam(selectedTeam);
+    } else if (
+      selectedTeam !== project.team &&
+      selectedStream === project.stream
+    ) {
+      setIsValidTeam(true);
+      setIsValidStream(true);
+      setTeam(selectedTeam);
+    }
+  };
+
+  useEffect(() => {
+    if (!selectedTeam) {
+      setIsValidTeam(false);
+    }
+  }, [selectedTeam]);
+
+  const [selectedLabels, setLabels] = useState(project.labels);
+  const [isValidLabels, setIsValidLabels] = useState(true);
+  const [labelError, setLabelError] = useState("");
+  const onLabelChange = labels => {
+    const labelsValid =
+      labels.length === 0
+        ? true
+        : labels.reduce((labelsValid, label) => {
+            return labelsValid && label.isKeyValid && label.isValueValid;
+          }, true);
+    setIsValidLabels(labelsValid);
+    if (!labelsValid) {
+      setLabelError(
+        "Invalid labels. Both key and value of a label must contain only alphanumeric and dash (-)"
+      );
+    }
+
+    if (
+      selectedTeam === project.team &&
+      selectedStream === project.stream &&
+      selectedLabels !== project.labels
+    ) {
+      setIsValidTeam(true);
+      setIsValidStream(true);
+      setIsValidLabels(labelsValid);
+      setTeam(selectedTeam);
+      setStream(selectedStream);
+    }
+
+    //deep copy
+    let newLabels = JSON.parse(JSON.stringify(labels));
+    newLabels = newLabels.map(element => {
+      delete element.isKeyValid;
+      delete element.isValueValid;
+      delete element.idx;
+      return element;
+    });
+
+    setLabels(newLabels);
+  };
+  return (
+    <>
+      <EuiFlexGroup direction="column">
```

Comment:
im not sure if this is related but you are using `EuiFormRow` without it being in a `EuiForm`, so i think the responsiveness is off